### ISSUE TITLE
[BUG] [Go] object nullable not working

### ISSUE THREAD

**Issue by hatappi** [OpenAPITools/openapi-generator#2119]
##### Description
output json using Go `struct`.
When there is no value in strcut, it may not be displayed in json.
In such a case I will use pointer.

However, if specify `nullable: true` for the object in the definition file, it will not be output with a pointer.
For integer and string pointers are given.

##### openapi-generator version
- 3.3.4
- 4.0.0-SNAPSHOT

##### OpenAPI declaration file content or url

**test.yaml**

```yaml
openapi: "3.0.6"
info:
  version: "1.0"
  title: test
paths:
  /test:
    get:
      summary: test
      responses:
        '200':
          description: OK
components:
  schemas:
    user:
      type: object
      properties:
        name:
          type: string
          nullable: true
        age:
          type: integer
          nullable: true
        address:
          type: object
          nullable: true
          properties:
            city:
              type: string
```

**model.mustache**

```
{{>partial_header}}
package {{packageName}}
{{#models}}
{{#imports}}
{{#-first}}
import (
{{/-first}}
	"{{import}}"
{{#-last}}
)
{{/-last}}
{{/imports}}
{{#model}}
{{#isEnum}}
{{#description}}
// {{{classname}}} : {{{description}}}
{{/description}}
type {{{classname}}} {{^format}}{{dataType}}{{/format}}{{#format}}{{{format}}}{{/format}}

// List of {{{name}}}
const (
	{{#allowableValues}}
	{{#enumVars}}
	{{^-first}}
	{{/-first}}
	{{{classname}}}{{name}} {{{classname}}} = "{{{value}}}"
	{{/enumVars}}
	{{/allowableValues}}
){{/isEnum}}{{^isEnum}}{{#description}}
// {{{description}}}{{/description}}
type {{classname}} struct {
{{#vars}}
{{^-first}}
{{/-first}}
{{#description}}
	// {{{description}}}
{{/description}}
	{{name}} {{#isNullable}}*{{/isNullable}}{{dataType}} `json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
{{/vars}}
}
{{/isEnum}}
{{/model}}
{{/models}}
```

##### Command line used for generation

```shell
docker run --rm \
  -v ${PWD}:/local \
  openapitools/openapi-generator-cli:v3.3.4 \
  generate \
  -i /local/test.yaml \
  -g go \
  -o /local/pkg \
  -t /local/templates
```

##### Steps to reproduce
1. Prepare `templates/model.mustache`
2. Save OpenAPI declaration file with test.yaml
3. run command
4. check pkg/model_user.go

**Comment by ackintosh** [comment]
Hi @hatappi ,

[`nullable` support for the Go generator ](https://github.com/OpenAPITools/openapi-generator/pull/1869) has been merged into master. please give it a try with `latest` docker image. 😉

**Comment by wing328** [comment]
```
openapi: "3.0.6"
info:
  version: "1.0"
  title: test
```

Btw, I don't think OAS 3.0.6 has released yet. The latest is "3.0.2": https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md

**Comment by hatappi** [comment]
@ackintosh @wing328 
Thank you for reply.
I tried `latest` docker image. `Image Id` is `e0c35c5a73f5`.
But the pointer did not attach to a struct.

**test.yaml**

```
openapi: "3.0.2"
info:
  version: "1.0"
  title: test
paths:
  /test:
    get:
      summary: test
      responses:
        '200':
          description: OK
components:
  schemas:
    user:
      type: object
      properties:
        name:
          type: string
          nullable: true
        age:
          type: integer
          nullable: true
        address:
          type: object
          nullable: true
          properties:
            city:
              type: string
```

**command**

```shell
docker run --rm \
  -v ${PWD}:/local \
  openapitools/openapi-generator-cli \
  generate \
  -i /local/test.yaml \
  -g go \
  -o /local/pkg
```

**generated model**

```go
/*
 * test
 *
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * API version: 1.0
 * Generated by: OpenAPI Generator (https://openapi-generator.tech)
 */

package openapi

type User struct {
	Name *string `json:"name,omitempty"`
	Age *int32 `json:"age,omitempty"`
	Address UserAddress `json:"address,omitempty"`
}
```

**Comment by kemokemo** [comment]
Thank you very much for reporting issue!
I am sorry that confirmation with Issue #1869 was insufficient.
I will investigate.

**Comment by kemokemo** [comment]
Hmm..
In the `fromProperty` method of the `DefaultCodegen.java`, `p.getNullable()` returns false for the `address` schema. [this link](https://github.com/OpenAPITools/openapi-generator/blob/cd1885d29e194f17640c60919f00a7a46a8745d9/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L1916)

@wing328 @ackintosh 
Sorry, could you please give me any advice?

**Comment by ackintosh** [comment]
Thanks @kemokemo . Let me check 👀

**Comment by ackintosh** [comment]
Ah 💡 It is due to [InlineModelResolver](https://github.com/OpenAPITools/openapi-generator/blob/046db19a8590f3557ae1e04be75792bbeb39cb69/modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java). 

https://github.com/OpenAPITools/openapi-generator/blob/f76dca84f542fe5d9354501f7f02142238fa562a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L888

[`InlineModelResolver#modelFromProperty()`](https://github.com/OpenAPITools/openapi-generator/blob/046db19a8590f3557ae1e04be75792bbeb39cb69/modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java#L519-L539) (which is called from `flatten()`) resolves inline model `user.address` to `UserAddress` but the resolved `UserAddress` model has no `nullable ` value. 😢

**Comment by ackintosh** [comment]
I'll file a PR to fix the missing `nullable`. 😉

**Comment by ackintosh** [comment]
@hatappi  The fix has been merged into master but the docker image `latest` has not updated yet. please wait a moment. 😌

**Comment by hatappi** [comment]
@ackintosh 
Thank you 😊 
I'm looking forward to it.

**Comment by ackintosh** [comment]
The docker image has been updated. I've verified with the image that the `UserAddress` is generated as pointer.

```diff
type User struct {
        Name *string `json:"name,omitempty"`
        Age *int32 `json:"age,omitempty"`
-        Address UserAddress `json:"address,omitempty"`
+        Address *UserAddress `json:"address,omitempty"`
}
```

Again thanks for reporting this issue. 👍

**Comment by hatappi** [comment]
@ackintosh 
I also confirmed that it was fixed by `246d3a3f904a` docker image 🎉 
Thank you for fixed.
